elastic/kibana
# [Console] Allow users to copy requests as specific languages

## Summary

This PR enhances `copy as` functionality from monaco console by leveraging the [@elastic/request-converter](https://www.npmjs.com/package/@elastic/request-converter?activeTab=code) package. This will allow the user to copy the devtools commands not only as `curl` but also as `JavaScript` and `Python`.

##### How to test

* Start up es and kibana
* Navigate to `Devtools`
* Verify that:
  * By default `Copy as curl` is selected
  * Code is copied correctly
  * Language can be changed while keeping a separate default language
  * When copying a request to a given language a loading spinner should show up indicating that there is a pending action going on (you can throttle the network connection through browser devtools in order to see this)
  * Selecting a default language stores the setting in localstorage
  * We retain the same old functionality in `ace editor`


https://github.com/user-attachments/assets/a251e9e1-50a7-47bb-bc25-3d8686a7b838



<details>
<summary>UI screenshots</summary>

![Screenshot 2024-08-09 at 10 46 48](https://github.com/user-attachments/assets/bba09d75-47ae-4460-99a0-dd31f0ead53e)
![Screenshot 2024-08-09 at 10 46 54](https://github.com/user-attachments/assets/120b0ab4-83d1-4f13-8fc8-8c0adb481810)
![Screenshot 2024-08-09 at 10 47 00](https://github.com/user-attachments/assets/b04fedc7-47ad-4712-a5d2-ebf648a97805)
![Screenshot 2024-08-09 at 10 47 11](https://github.com/user-attachments/assets/ae6e6a1a-75e8-434a-b5d6-e5cb716565f6)

</details>

<details>
<summary>Tooltip screenshots</summary>

![Screenshot 2024-08-09 at 10 46 33](https://github.com/user-attachments/assets/e2877eac-d98c-455f-8d2f-455b88ea93e8)
![Screenshot 2024-08-09 at 10 46 40](https://github.com/user-attachments/assets/ebb93548-1b39-4e4a-8cfb-fcc3ac9d92b9)


</details>
